languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -113,22 +106,32 @@ private void handleCall(String attrName, JSONArray jArgs, JSONObject polarInstan
                     if (host.hasInstance(id)) {
                         throw new Exceptions.DuplicateInstanceRegistrationError(id);
                     }
-                    String clsName = data.getJSONObject("instance").getString("tag");
-                    JSONObject jFields = data.getJSONObject("instance").getJSONObject("fields").getJSONObject("fields");
-                    host.makeInstance(clsName, host.polarDictToJava(jFields), id);
+                    JSONObject constructor = data.getJSONObject("constructor").getJSONObject("value");
+                    JSONArray initargs;
+                    if (constructor.has("InstanceLiteral")) {
+                        // Keyword initargs are not supported in Java.
+                        className = constructor.getJSONObject("InstanceLiteral").getString("tag");
+                        throw new Exceptions.InstantiationError(className);
```

Comment:
This is what I was envisioning to allow support of both kwargs and positional args, but now I'm confused. Are kwarg constructors still supported for other languages?

Reply:
Yes.